uwblueprint/website-bp
# Token auth setup

[[FE] Dashboard - Authentication & Saving Token](https://www.notion.so/uwblueprintexecs/FE-Dashboard-Authentication-Saving-Token-f537cc9fb44c4c0e92df387635475317)

### Implementation description
- Removed UserContext because we are now handling auth information directly using the access tokens and not the UserContext.
- Edit the login in the BFF layer to set the access and refresh token after login query
- Changed ProtectedRoute component to use access token for verification on a user role



### Steps to test
NOTE: In order to test, you will have to link this up with the `Token auth setup` backend branch.
1. Go to http://localhost:3000/admin and you should be redirected to http://localhost:3000/admin/login
2. Login at http://localhost:3000/admin/login and notice that your access and refresh tokens are populated in local storage (Under application of google dev tools)
3. Go back to http://localhost:3000/admin and notice that you should see the dashboard (Assuming you logged in with an admin role user)



### What should reviewers focus on?
- Focus on `website-bp/components/context/ProtectedRoute.tsx` as I'm not really sure on whether the way I handle updating react state and return statement is "clean".  Maybe it's possible to reduce the code to use less usestate variables and handle url redirecting to localhost:300/admin/login better

### Checklist
